### DISCUSSION TITLE
/login/magic-link doesn't redirect to /auth/a/show

### DISCUSSION THREAD

**Discussion by GeorgKott** [codeigniter4/shield#1006] [Q&A] [answered]
In LoginController for example i see this code in the method loginView

https://github.com/codeigniter4/shield/blob/develop/src/Controllers/LoginController.php#L42

```php
        $authenticator = auth('session')->getAuthenticator();
        
        // If an action has been defined, start it up.
        if ($authenticator->hasAction()) {
            return redirect()->route('auth-action-show');
        }
```

the same in RegisterController in registerView.

But in MagicLinkController only in verify method, not in loginView.
https://github.com/codeigniter4/shield/blob/develop/src/Controllers/MagicLinkController.php#L60

is it a wrong behavior? Or i have a mistake im my head. 
Thank you

**Comment by kenjis** [comment]
What's your issue?

**Reply by GeorgKott** [reply]
Is it ok that "/login/magic-link" doesn't redirect to "/auth/a/show" ?

**Reply by kenjis** [reply, marked as answer]
I don't see any problem. What's wrong with it?

**Reply by pippuccio76** [reply]
Magick link shouldn't it redirect to change password?

**Reply by datamweb** [reply]
If you need it, you can use the [magiclogin](https://shield.codeigniter.com/references/events/?h=#magiclogin) Or [session-notification](https://shield.codeigniter.com/references/magic_link_login/#session-notification).

**Reply by pippuccio76** [reply]
why there isn't a page  to change password or email or username ? I must implements all ?

**Reply by kenjis** [reply]
> why there isn't a page to change password or email or username ?

Because no one has sent a Pull Request with such pages.
